scipy/optimize/tests/test__optimize.py

```diff
@@ -0,0 +1,47 @@
+import numpy as np
+
+import pytest
+from numpy.testing import assert_allclose, assert_
+
+from scipy.optimize.optimize import MemoizeJac
+
+class FunctionWithGradient(object):
```

Comment:
Could you also add a test from the original issue please?

Reply:
I adapted the example from the original issue into an additional test. Should I reference the original issue in the test?

Reply:
Yes, a comment like "tests changes made in response to GHXXXX" would do.

Reply:
I added the comment.